qa/workunits/rbd/cli_generic.sh

```diff
@@ -1358,6 +1358,52 @@ test_mirror_snapshot_schedule() {
     ceph osd pool rm rbd2 rbd2 --yes-i-really-really-mean-it
 }
 
+test_mirror_snapshot_schedule_recovery() {
```

Comment:
Couple of notes regarding this test:
- Maybe we want to combine this with test_mirror_snapshot_schedule? But I didn't do that for now considering that https://tracker.ceph.com/issues/58490 causes the test_mirror_snapshot_schedule to fail

- This test uncovered a bug in the changes I made to  `rbd_support.schedule.LevelSpec.from_name`, where I replaced `module.rados` with `handler.rados` to allow the scheduler handlers to only use handler's RADOS client.
I hit the following error,
```
2023-02-04T20:22:17.389-0500 7f9035f8a640  0 [rbd_support ERROR root] Fatal runtime error:
Traceback (most recent call last):
File "/home/rraja/git/ceph/src/pybind/mgr/rbd_support/module.py", line 42, in wrapper
    return func(self, *args, **kwargs)
File "/home/rraja/git/ceph/src/pybind/mgr/rbd_support/module.py", line 102, in mirror_snapshot_schedule_add
    spec = LevelSpec.from_name(
 File "/home/rraja/git/ceph/src/pybind/mgr/rbd_support/schedule.py", line 125, in from_name
    raise ValueError("{} is not an RBD pool".format(pool_name))
ValueError: rbd3 is not an RBD pool
```
This was because the pool ID of `rbd3` fetched using `handler.rados.pool_lookup(pool_name)` (pool ID was 8) was different from the pool ID of `rbd3` in the output of `get_rbd_pools(handler.module)` (pool ID was 9) .  I'm not sure why this happens? So in my local testing I reverted the changes I made to `rbd_support.schedule.LevelSpec.from_name` to continue using  `module.rados`, which fixed this issue.

Reply:
I suppose it is because now your (not module's) rados client does not wait for the latest osd map, and you use the ID (8) of rbd3 pool that was created (and deleted) for the previous test.

Please look at the `with_latest_osdmap` decorator in `module.py` and how it is used to make sure all commands that need the latest osdmap wait for it. You need this for your rados instances too.

Reply:
> This was because the pool ID of `rbd3` fetched using `handler.rados.pool_lookup(pool_name)` (pool ID was 8) was different from the pool ID of `rbd3` in the output of `get_rbd_pools(handler.module)` (pool ID was 9) .

Is this reproducible?  Always or only if `test_trash_purge_schedule_recovery` (or anything else) is executed before `test_mirror_snapshot_schedule_recovery`?  I suspect the root cause boils down to the fact that different RADOS instances may have (be aware of) different versions of osdmap.  It's not uncommon for a specific RADOS instances to be a little "behind" and you are re-creating a pool with same name here.  Also notice that:
- `get_rbd_pools` operates on the osdmap that is returned directly by the manager framework (it's cached)
- `with_latest_osdmap` in `src/pybind/mgr/rbd_support/module.py` syncs up with the latest osdmap using the module-wide RADOS instance

Reply:
Ah, it looks like @trociny beat me to it -- my comments sat there waiting for "Finish review" to be clicked ;)

Reply:
> I suppose it is because now your (not module's) rados client does not wait for the latest osd map, and you use the ID (8) of rbd3 pool that was created (and deleted) for the previous test.
> 
> Please look at the `with_latest_osdmap` decorator in `module.py` and how it is used to make sure all commands that need the latest osdmap wait for it. You need this for your rados instances too.

Thanks @trociny ! I missed making the rados instances wait for latest osdmap when connecting and when a command was rbd_support module command handled. I did that and the tests passed.

Reply:
> > This was because the pool ID of `rbd3` fetched using `handler.rados.pool_lookup(pool_name)` (pool ID was 8) was different from the pool ID of `rbd3` in the output of `get_rbd_pools(handler.module)` (pool ID was 9) .
> 
> Is this reproducible? Always or only if `test_trash_purge_schedule_recovery` (or anything else) is executed before `test_mirror_snapshot_schedule_recovery`? 

I didn't hit this bug in my local testing. I assumed  I'll hit it when running this PR in teuthology.

> I suspect the root cause boils down to the fact that different RADOS instances may have (be aware of) different versions of osdmap. It's not uncommon for a specific RADOS instances to be a little "behind" and you are re-creating a pool with same name here. Also notice that:
> 
> * `get_rbd_pools` operates on the osdmap that is returned directly by the manager framework (it's cached)
> * `with_latest_osdmap` in `src/pybind/mgr/rbd_support/module.py` syncs up with the latest osdmap using the module-wide RADOS instance

Thanks for the explanation. I dug further and saw that `get_rbd_pools` eventually calls into `pymodules->cacheable_get_python` . Yes, I missed making the RADOS instances of the handlers wait for the latest osdmap.